server/tables/handler.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tables
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// Handler is an interface that controls how data is represented for some table.
+type Handler interface {
+	// Name returns the name of the table.
+	Name() string
+	// RowIter returns a sql.RowIter that returns the rows of the table.
+	RowIter(ctx *sql.Context) (sql.RowIter, error)
+	// Schema returns the table's schema.
+	Schema() sql.PrimaryKeySchema
+}
+
+// handlers is a map from the schema name, to the table name, to the handler.
+var handlers = map[string]map[string]Handler{}
```

Comment:
Seems like static declaration should be fine for this, no? Just a big `var` declaration with a static map of constants and handler functions. Better to have this entire thing declared statically in one place than to have it filled in piecemeal by a bunch of different places. There's a single cycle in the type assertion of pg_sequence, but if you put these into a static `map[string]Handler` in this package that takes care of the assertion. Or you could always make a common package for both these places to import, up to you.